Fix the Travis CI builds

The builds currently all fail with the following log messages:

<pre>
node-pre-gyp WARN Using needle for node-pre-gyp https download 
node-pre-gyp WARN Tried to download(404): https://github.com/node-gfx/node-canvas-prebuilt/releases/download/v2.4.1/canvas-v2.4.1-node-v72-linux-glibc-x64.tar.gz 
node-pre-gyp WARN Pre-built binaries not found for canvas@2.4.1 and node@12.1.0 (node-v72 ABI, glibc) (falling back to source compile with node-gyp) 
Package pangocairo was not found in the pkg-config search path.
Perhaps you should add the directory containing `pangocairo.pc'
to the PKG_CONFIG_PATH environment variable
No package 'pangocairo' found
gyp: Call to 'pkg-config pangocairo --libs' returned exit status 1 while in binding.gyp. while trying to load binding.gyp
gyp ERR! configure error 
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (/home/travis/.nvm/versions/node/v12.1.0/lib/node_modules/npm/node_modules/node-gyp/lib/configure.js:345:16)
gyp ERR! stack     at ChildProcess.emit (events.js:196:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:256:12)
gyp ERR! System Linux 4.4.0-101-generic
gyp ERR! command "/home/travis/.nvm/versions/node/v12.1.0/bin/node" "/home/travis/.nvm/versions/node/v12.1.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "configure" "--fallback-to-build" "--module=/home/travis/build/mozilla/pdf.js/node_modules/canvas/build/Release/canvas.node" "--module_name=canvas" "--module_path=/home/travis/build/mozilla/pdf.js/node_modules/canvas/build/Release" "--napi_version=4" "--node_abi_napi=napi" "--napi_build_version=0" "--node_napi_label=node-v72"
gyp ERR! cwd /home/travis/build/mozilla/pdf.js/node_modules/canvas
gyp ERR! node -v v12.1.0
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok 
node-pre-gyp ERR! build error 
node-pre-gyp ERR! stack Error: Failed to execute '/home/travis/.nvm/versions/node/v12.1.0/bin/node /home/travis/.nvm/versions/node/v12.1.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js configure --fallback-to-build --module=/home/travis/build/mozilla/pdf.js/node_modules/canvas/build/Release/canvas.node --module_name=canvas --module_path=/home/travis/build/mozilla/pdf.js/node_modules/canvas/build/Release --napi_version=4 --node_abi_napi=napi --napi_build_version=0 --node_napi_label=node-v72' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/home/travis/build/mozilla/pdf.js/node_modules/node-pre-gyp/lib/util/compile.js:83:29)
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:196:13)
node-pre-gyp ERR! stack     at maybeClose (internal/child_process.js:1000:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:267:5)
node-pre-gyp ERR! System Linux 4.4.0-101-generic
node-pre-gyp ERR! command "/home/travis/.nvm/versions/node/v12.1.0/bin/node" "/home/travis/build/mozilla/pdf.js/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build"
node-pre-gyp ERR! cwd /home/travis/build/mozilla/pdf.js/node_modules/canvas
node-pre-gyp ERR! node -v v12.1.0
node-pre-gyp ERR! node-pre-gyp -v v0.11.0
node-pre-gyp ERR! not ok 
Failed to execute '/home/travis/.nvm/versions/node/v12.1.0/bin/node /home/travis/.nvm/versions/node/v12.1.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js configure --fallback-to-build --module=/home/travis/build/mozilla/pdf.js/node_modules/canvas/build/Release/canvas.node --module_name=canvas --module_path=/home/travis/build/mozilla/pdf.js/node_modules/canvas/build/Release --napi_version=4 --node_abi_napi=napi --napi_build_version=0 --node_napi_label=node-v72' (1)
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.4 (node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.4: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! canvas@2.4.1 install: `node-pre-gyp install --fallback-to-build`
npm ERR! Exit status 1
</pre>

This is because https://github.com/node-gfx/node-canvas-prebuilt/releases does not have a version for the Node version 72 ABI. This is requested in https://github.com/node-gfx/node-canvas-prebuilt/issues/78 though, so we might wait or try to get the fallback compilation working somehow.

--- a/package.json
+++ b/package.json
@@ -10,7 +10,7 @@
     "acorn": "^6.1.1",
     "autoprefixer": "^9.5.1",
     "babel-loader": "^8.0.5",
-    "canvas": "^2.4.1",
+    "canvas": "^2.5.0",
     "core-js": "^2.6.5",
     "escodegen": "^1.11.1",
     "eslint": "^5.16.0",

--- a/test/unit/parser_spec.js
+++ b/test/unit/parser_spec.js
@@ -99,7 +99,17 @@ describe('parser', function() {
         for (const number of numbers) {
           const input = new StringStream(number);
           const lexer = new Lexer(input);
-          expect(lexer.getNumber()).toEqual(parseFloat(number));
+
+          const result = lexer.getNumber(), expected = parseFloat(number);
+
+          if (result !== expected && Math.abs(result - expected) < 1e-15) {
+            console.error(`Fuzzy matching "${result}" with "${expected}" to ` +
+                          'work-around rounding bugs in Chromium browsers.');
+
+            expect(true).toEqual(true);
+            continue;
+          }
+          expect(result).toEqual(expected);
         }
       });
 

